chapter5/findwithflag.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func excludeNames(name string, exclude string) bool {
+	if exclude == "" {
+		return false
+	}
+	if filepath.Base(name) == exclude {
+		return true
+	}
+	return false
+}
+
+func excludeExtensions(name string, extension string) bool {
+	if extension == "" {
+		return false
+	}
+	basename := filepath.Base(name)
+	s := strings.Split(basename, ".")
+	length := len(s)
+	basenameExtension := s[length-1]
+	if basenameExtension == extension {
+		return true
+	}
+	return false
+}
+
+func main() {
+
+	minusS := flag.Bool("s", false, "Sockets")
+	minusP := flag.Bool("p", false, "Pipes")
+	minusSL := flag.Bool("sl", false, "Symbolic Links")
+	minusD := flag.Bool("d", false, "Directories")
+	minusF := flag.Bool("f", false, "Files")
+	minusX := flag.String("x", "", "Files")
+	minusEXT := flag.String("ext", "", "Extensions")
+
+	flag.Parse()
+	flags := flag.Args()
+
+	printAll := false
+	if *minusS && *minusP && *minusSL && *minusD && *minusF {
+		printAll = true
+	}
+
+	if !(*minusS || *minusP || *minusSL || *minusD || *minusF) {
+		printAll = true
+	}
+
+	if len(flags) == 0 {
```

Comment:
why check for len? if you have to treat flags as positional arguments then wh use flag package at all? then you can use os.ARGS only

Reply:
since we are using flag for flags anyways  and it has abstration over os.Args which does the same job

Reply:
You mean using flag you can get positional as well as names arguments at the same time?

Reply:
yes using flag.Args we can take positional arguments only difference between this and os.Args is its indexing starts from zero